### ISSUE TITLE
[Due for payment 2026-04-14] [$1000] Screen Reader: Workspace Category: The accordion button is announced without a state

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#76929]
## Action Performed:
Prerequisite:
- User is logged in

1. Using Chrome+JAWS, open the URL: https://new.expensify.com/
2. Tab to focus on the required workspace and press Enter to activate it
3. Once the Workspace menu is displayed, tab to focus on the "Categories" menu item and press Enter to activate it
4. Tab to focus the More button and observe the screen reader announcement.

## Expected Result:
The accordion button should communicate its expanded or collapsed state to assistive technologies. When expanded, the button should indicate “expanded”, and when collapsed, it should indicate “collapsed”.

## Actual Result:
The expanded collapsed state of the accordion button "more" is not announced by the screen reader upon activation. When focused the screen reader only announces the role and label of the control without indicating whether it is expanded or collapsed. As a result, users relying on assistive technologies cannot determine the current state of the accordion content.

HTML:
<button aria-label="" role="button" data-selection-scrapper-hidden-element="true" data-tag="pressable" tabindex="0" class="css-g5y9jx r-1otgn73 r-1loqt21 r-1777fci r-1mdbw0j r-wk8lta r-1awozwy r-1iofnf8 r-12he9u2 r-18u37iz r-9ffhgg r-lrvibr" id="" type="button" data-component="GenericPressable" data-element="Pressable" data-source-file="BaseGenericPressable.tsx" style="border-radius: 100px; min-height: 40px; min-width: 40px; padding-left: 16px; padding-right: 12px; height: 40px; background-color: rgb(230, 225, 218);"><div class="css-g5y9jx r-1wtj0ep r-18u37iz r-dnmrzs"><div class="css-g5y9jx r-1awozwy r-18u37iz r-1wbh5a2"><div dir="auto" data-selection-scrapper-hidden-element="true" class="css-146c3p1 r-dnmrzs r-1udh08x r-1udbk01 r-3s2u2q r-1iln25a" style="color: rgb(0, 46, 34); font-size: 13px; text-align: center; font-family: &quot;Expensify Neue&quot;, &quot;Windows Segoe UI Emoji&quot;, &quot;Noto Color Emoji&quot;, &quot;Custom Emoji Font&quot;; font-style: normal; font-weight: 700; pointer-events: none; flex-shrink: 1; padding-bottom: 1px;">More</div></div><div class="css-g5y9jx r-1777fci r-13hce6t"><div class="css-g5y9jx" data-testid=""><svg xmlns="[http://www.w3.org/2000/svg"](http://www.w3.org/2000/svg%22) xml:space="preserve" viewBox="0 0 20 20" width="16" height="16" hovered="false" pressed="false" fill="#A2A9A3"><path d="M18.4 5.6c.8.8.8 2 0 2.8L10 16.8 1.6 8.4c-.8-.8-.8-2 0-2.8s2-.8 2.8 0l5.6 5.6 5.6-5.6c.8-.8 2-.8 2.8 0" style="fill-rule: evenodd; clip-rule: evenodd;"></path></svg></div></div></div></button>

## Area issue was found in:
Workspace Category

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Add the "aria-expanded" attribute to the accordion button to indicate its current state.

Refer:
https://www.w3.org/WAI/ARIA/apg/patterns/accordion/

## Environment(s)
Windows Windows 11 - Chrome

## Screenshots/Videos
https://github.com/user-attachments/assets/f95d048f-8da4-4834-b55b-672b9a0d693f

<img width="1919" height="966" alt="Image" src="https://github.com/user-attachments/assets/3192dad1-3e52-44f1-a130-0148c2259b39" />

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6996806 
**Issue reported by:** NA
**Slack conversation:** NA

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~022013102921460848748</li>
        <li>Upwork Job ID: 2013102921460848748</li>
        <li>Last Price Increase: 2026-04-14</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by rushatgabhane** [comment]
Edit: my bad. wrong issue. 

This is being fixed

**Comment by rushatgabhane** [comment]
I couldn't fix this one using AI. 

Looking for proposals

**Comment by mallenexpensify** [comment]
👀 above plz @QichenZhu . Thx

**Comment by QichenZhu** [comment]
@daledah, could you test your proposal and record a video? It's not working for me. Thanks!

**Comment by QichenZhu** [comment]
@daledah thanks for the update. But I'm still seeing two problems:

1. Even non-dropdown buttons like Invite are announced as "collapsed"
2. After opening the More menu and pressing ESC to close it, pressing Enter doesn't expand it again

**Comment by QichenZhu** [comment]
> [@QichenZhu](https://github.com/QichenZhu) could you please share your screenshots? And I can use that to test on my device.

@daledah here's a recording. You can hear it doesn't announce the state on iOS with your solution.

https://github.com/user-attachments/assets/433cf7c7-0f05-4b69-8ac9-ec5b71166fa4

**Comment by QichenZhu** [comment]
@dangrous, based on [@daledah's](https://github.com/Expensify/App/issues/76929#issuecomment-3836560934) and [my](https://github.com/Expensify/App/issues/76929#issuecomment-3822271670) testing, it announces when expanded but not when collapsed (so the user doesn't know it's expandable).

**Comment by rushatgabhane** [comment]
Seems like upstream bug?

**Comment by dangrous** [comment]
sweet! what do we need to move forward?

**Comment by MelvinBot** [comment]
🤖 **Cannot proceed:** `QichenZhu` does not have write access to the target repository (`Expensify/App`). MelvinBot will not create branches or PRs on repos that the requesting user cannot access themselves.

**Comment by mallenexpensify** [comment]
@QichenZhu @dangrous can either of you plz confirm @daledah 's comment above so I can pay 'em?  

> assigned to this issue, and Melvin’s PR also used the solution we had previously discussed.

If so, it looks like it'd be 100%, correct?

**Comment by QichenZhu** [comment]
> [@QichenZhu](https://github.com/QichenZhu) [@dangrous](https://github.com/dangrous) can either of you plz confirm [@daledah](https://github.com/daledah) 's comment above so I can pay 'em?

@mallenexpensify I can confirm @daledah opened a PR and requested review, and Melvin's solution built on @daledah's proposal.

**Comment by QichenZhu** [comment]
# BugZero Checklist:

- [x] **[Contributor]** The offending PR and associated issue have been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to the comment on the PR: N/A.
    Link to the comment on the Issue: N/A.

    Reported [an issue](https://github.com/facebook/react-native/issues/56296) and opened [a PR](https://github.com/facebook/react-native/pull/56430) upstream.

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: N/A.

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [x] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue: https://github.com/Expensify/Expensify/issues/625160

## Regression Test Proposal
### Precondition:


- Enable the screen reader

### Test:


1. Navigate to a workspace's Categories page
2. Focus the More button
3. Verify the screen reader announces the button as collapsed

Do we agree 👍 or 👎

<details><summary>Zapier Logs</summary>
Run ID: <a href="https://zapier.com/editor/203475862/run/00040eee-1443-a2fc-3336-b1a59b2da094/run-details">00040eee-1443-a2fc-3336-b1a59b2da094</a>
</details>

**Comment by QichenZhu** [comment]
@mallenexpensify @roryabraham, could you adjust the compensation based on https://github.com/Expensify/App/pull/86499#discussion_r3018127085?

**Comment by mallenexpensify** [comment]
### Payment Summary
Contributor+: @QichenZhu  due $1000 via NewDot
Contributor+: @daledah  due $1000 via NewDot

^ Test case created

Thx

**Comment by trjExpensify** [comment]
approved $1,000 for @QichenZhu

**Comment by quinthar** [comment]
Approved $1000 for @daledah
